pandas/core/arrays/timedeltas.py

```diff
@@ -551,47 +505,51 @@ def __rfloordiv__(self, other):
             raise TypeError("Cannot divide {typ} by {cls}"
                             .format(typ=dtype, cls=type(self).__name__))
 
+    @ops.unpack_and_defer
     def __mod__(self, other):
         # Note: This is a naive implementation, can likely be optimized
-        if isinstance(other, (ABCSeries, ABCDataFrame, ABCIndexClass)):
-            return NotImplemented
-
-        other = lib.item_from_zerodim(other)
         if isinstance(other, (timedelta, np.timedelta64, Tick)):
             other = Timedelta(other)
+            if other is NaT:
+                # convert back to something that floordiv will recognize
+                #  as timedelta-like
+                other = np.timedelta64('NaT', 'ns')
```

Comment:
This and the copy/pasted changes below are not central to this PR, should be separated out (and tested)